### PR TITLE
3105: Varnish 4 vcl file

### PR DESCRIPTION
**Pull request by cableman**
This PR adds a configuration file for varnish 4.0 that can be used with DDB CMS.

See https://platform.dandigbib.org/issues/3105

### CODE REVIEW

File: ding2_v4.vcl [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+  # We need this to cache 404s, 301s, 500s. Otherwise, depending on backend but
+  # definitely in Drupal's case these responses are not cacheable by default.
+  if (beresp.status == 404 || beresp.status == 301 || beresp.status == 500) {
+    set beresp.ttl = 10m;
+  }
+
+  if (beresp.http.x-drupal-cache == "HIT") {
+    unset beresp.http.Cache-Control;
+    set beresp.http.Cache-Control = "public,max-age=21600";
+  }
```
Comment (kasperg) [inline ding2_v4.vcl]: This always seems to cache pages which are also in Drupals cache with a hardcoded max-age. Why is this?

**Comment by cableman** [comment]
Right, added the vlc_error function.

File: ding2_v4.vcl (hunk 1 above)
Comment (kasperg) [reply ding2_v4.vcl]: @cableman: Where do you see this in [the existing `ding2.vcl`](https://github.com/ding2/ding2/blob/master/ding2.vcl)?

Comment (cableman) [reply ding2_v4.vcl]: Right removed it... can't remember why!

**Comment by kasperg** [comment]
I have rebuilt this multiple times. The build keeps failing with something unrelated to the change at hand. Consequently I will merge this despite the error.

**Pull request merged**